File: Course_1-Algorithmic_Toolbox/Week-2/Excercise_Challenges/4_least_common_multiple/lcm.cpp
#include <iostream>
using namespace std;

long long lcm_naive(int a, int b)
{
    for(long long l = 1; l <= (long long) a * b; ++l)
        if(l % a == 0 && l % b == 0)
            return l;

    return (long long) a * b;
}

int gcd_euclid(int a, int b)
{
    if(b == 0)
        return a;
    return gcd_euclid(b, a % b);
}

long long lcm_fast(int a, int b)
{
    return ((long long)a * b) / gcd_euclid(a, b);
}

int main(void)
{
    int a, b;
    cin >> a >> b;

    //cout << lcm_naive(a, b) << endl;
    cout << lcm_fast(a, b) << endl;

    return 0;
}

File: Course_1-Algorithmic_Toolbox/Week-2/Excercise_Challenges/4_least_common_multiple/lcm.py
def lcm_naive(a, b):
    for l in range(1, a*b + 1):
        if l % a == 0 and l % b == 0:
            return l

    return a*b

def gcd_euclid(a, b):
    if b == 0:
        return a
    else:
        return gcd_euclid(b, a % b)

def lcm_fast(a, b):
    return int((a * b) / gcd_euclid(a, b))

if __name__ == '__main__':
    a, b = map(int, input().split())
    print(lcm_fast(a, b))